teachertool/src/hooks/useJsonDocRequest.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from "react";
+import { RequestStatus } from "../types";
+import { fetchJsonDocAsync } from "../services/backendRequests";
+
+export function useJsonDocRequest<T>(
+    url: string,
+    setStatus: (status: RequestStatus) => void,
+    setJson: (data: T) => void
+) {
+    const [fetching, setFetching] = useState<boolean>();
+
+    useEffect(() => {
+        if (!fetching) {
+            setFetching(true);
+            setStatus("loading");
+            Promise.resolve()
+                .then(async () => {
+                    const json = await fetchJsonDocAsync(url);
+                    if (!json) {
+                        setStatus("error");
+                    } else {
+                        setStatus("success");
+                        setJson(json as T);
+                    }
+                })
+                .catch(() => {
+                    setStatus("error");
+                });
+        }
+    }, [fetching]);
```

Comment:
Before, when the state variable was `status`, there were no dependencies in the `useEffect`, why did this change?

Reply:
While reworking the function I noticed I had neglected to include this dependency. As it was, the value of `fetching` was captured and would never change within useEffect handler. Probably harmless, but the code is corrected now.